### PR TITLE
Internationalise application

### PR DESCRIPTION
**Pull request by miurahr**
Here is a follow-up PR that split  logic from #129 

- Select language from Accept-Language header.
- Use language preference for login user.
- Accept locale url parameter when exist.

Signed-off-by: Hiroshi Miura <miurahr@linux.com>

### CODE REVIEW

File: config/initializers/locale.rb [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# Whitelist locales available for the application
+I18n.available_locales = %i[en ja]
```
Comment (MikeMcQuaid) [inline config/initializers/locale.rb]: Does this need any sort of translation file to not 💥? Should we hold off adding `ja` until the translation is added?

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -1,7 +1,41 @@
...
+    I18n.locale =
+      if user_signed_in?
+        current_user.language
+      else
+        determine_language
+      end
+  end
+
+  def default_url_options(*)
+    { locale: I18n.locale == I18n.default_locale ? nil : I18n.locale }
```
Comment (MikeMcQuaid) [inline app/controllers/application_controller.rb]: Mind doing `(I18n.locale == I18n.default_locale if I18n.locale)`, just for a slight readability tweak? Took me a minute to figure this out.

File: config/initializers/locale.rb (hunk 1 above)
Comment (miurahr) [inline config/initializers/locale.rb]: You are  all right. Ill fix it.

File: app/controllers/application_controller.rb (hunk 2 above)
Comment (miurahr) [inline app/controllers/application_controller.rb]: When user prefer English that is a default locale, we do not add ?locale=en parameter.

Comment (MikeMcQuaid) [inline app/controllers/application_controller.rb]: What about something like:

```ruby
def default_url_options(*)
  return unless I18n.locale
  return if I18n.locale == I18n.default_locale
  { locale: I18n.locale }
end
```

**Pull request merged**